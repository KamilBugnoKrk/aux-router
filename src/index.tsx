import * as React from 'react'
import {
  useHistory,
  BrowserRouter,
  Route,
  useRouteMatch
} from 'react-router-dom'

interface AuxLinkProps {
  componentName?: string
  componentValue?: string
  description: string
}

export const AuxLink = (props: AuxLinkProps) => {
  const history = useHistory()
  const match = useRouteMatch()

  function getComponentName() {
    if (match.url.lastIndexOf('(') < 0) {
      throw 'AuxLink should have componentName if it is not within AuxRoute element!'
    }
    return match.url.substring(match.url.lastIndexOf('(') + 1)
  }

  function onClick(): void {
    let oldPath = history.location.pathname
    let componentName = props.componentName
      ? props.componentName
      : getComponentName()

    let regex: RegExp
    if (componentName.indexOf(')') >= 0) {
      componentName = componentName.replace(')', '')
      regex = new RegExp(`(${componentName})`)
    } else {
      regex = new RegExp(`(${componentName}\\/\\w+)`)
    }
    let foundComponent = oldPath.match(regex)

    if (foundComponent) {
      history.push(
        oldPath.replace(regex, `${componentName}/${props.componentValue}`)
      )
    } else {
      history.push(oldPath + `(${componentName}/${props.componentValue})`)
    }
  }

  return <button onClick={onClick}>{props.description}</button>
}

export const AuxRouter = (props: any) => {
  return <BrowserRouter> {props.children}</BrowserRouter>
}

interface AuxRouteProps {
  component: any
  componentName?: string
  componentValue?: string
}

export const AuxRoute = (props: AuxRouteProps) => {
  const match = useRouteMatch()

  const getPath = (): string => {
    if (!props.componentValue) {
      return `*\\(${props.componentName}`
    } else if (match.url != '/') {
      debugger
      return `*${match.url.replaceAll('(', `\\(`).replaceAll(')', `\\)`)}/${
        props.componentValue
      }\\)*`
    } else {
      return `*\\(${props.componentName}/${props.componentValue}\\)*`
    }
  }

  return <Route path={getPath()} component={props.component} />
}
