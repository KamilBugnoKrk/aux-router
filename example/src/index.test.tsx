import React from 'react'
import renderer from 'react-test-renderer'
import { createMemoryHistory } from 'history'
import { AuxLink, AuxRoute, AuxMainLink, AuxMainRoute } from './../../src/index'
import Animals, { Type } from './Animals'
import { Cats, Dogs, Parrots } from './MockedData'
import { Router } from 'react-router-dom'
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer'
import Modal from '@material-ui/core/Modal'
import Birds from './Birds'

const advancedExample = (
  <>
    <nav>
      <AuxMainLink activeClassName='selected' path='/cats' description='Cats' />
      <AuxMainLink activeClassName='selected' path='/dogs' description='Dogs' />
      <AuxMainLink
        activeClassName='selected'
        path='/birds'
        description='Birds'
      />
    </nav>
    <AuxMainRoute path='/cats'>
      <Animals type={Type.Cat} animals={Cats} name='Cats' />
    </AuxMainRoute>
    <AuxMainRoute path='/dogs'>
      <Animals type={Type.Dog} animals={Dogs} name='Dogs' />
    </AuxMainRoute>
    <AuxMainRoute path='/birds' component={Birds} />
    {Dogs.map((animal, index) => {
      return (
        <AuxRoute componentName='panel-dog' componentValue={animal.name}>
          <SwipeableDrawer
            anchor='right'
            variant='persistent'
            open={true}
            onClose={() => {}}
            onOpen={() => {}}
            key={`${animal.name}-${index}`}
          >
            <div className='panel'>
              <img src={animal.image} alt={animal.name} />
              <p>Name: {animal.name}</p>
              <p>Age: {animal.age}</p>
              <AuxLink componentName='panel-dog' description='Close' />
            </div>
          </SwipeableDrawer>
        </AuxRoute>
      )
    })}
    {Parrots.map((animal, index) => {
      return (
        <AuxRoute componentName='panel-parrot' componentValue={animal.name}>
          <SwipeableDrawer
            anchor='bottom'
            variant='persistent'
            open={true}
            onClose={() => {}}
            onOpen={() => {}}
            key={`${animal.name}-${index}`}
          >
            <div className='panel'>
              <img src={animal.image} alt={animal.name} />
              <p>Name: {animal.name}</p>
              <p>Age: {animal.age}</p>
              <AuxLink componentName='panel-parrot' description='Close' />
            </div>
          </SwipeableDrawer>
        </AuxRoute>
      )
    })}
    {Cats.map((animal, index) => {
      return (
        <AuxRoute componentName='modal-cat' componentValue={animal.name}>
          <Modal open={true} onClose={() => {}} key={`${animal.name}-${index}`}>
            <div className='modal'>
              <img src={animal.image} alt={animal.name} />
              <p>Name: {animal.name}</p>
              <p>Age: {animal.age}</p>
              <AuxLink componentName='modal-cat' description='Close' />
            </div>
          </Modal>
        </AuxRoute>
      )
    })}
  </>
)

it('Advanced example /cats(modal/)(panel/Bean)', () => {
  const history = createMemoryHistory()
  const route = '/cats(modal/)(panel/Bean)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /cats(modal/)(panel/Bean)', () => {
  const history = createMemoryHistory()
  const route = '/cats(modal/)(panel/Bean)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /cats(modal/Bella)(panel/)', () => {
  const history = createMemoryHistory()
  const route = '/cats(modal/Bella)(panel/)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /dogs(modal/Bella)(panel/)', () => {
  const history = createMemoryHistory()
  const route = '/dogs(modal/Bella)(panel/)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /dogs(modal/)(panel/Bean)', () => {
  const history = createMemoryHistory()
  const route = '/dogs(modal/)(panel/Bean)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /dogs', () => {
  const history = createMemoryHistory()
  const route = '/dogs'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /cats', () => {
  const history = createMemoryHistory()
  const route = '/cats'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /cats(modal/Kitty)(panel/Bean)', () => {
  const history = createMemoryHistory()
  const route = '/cats(modal/Kitty)(panel/Bean)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /dogs(modal/Kitty)(panel/Bean)', () => {
  const history = createMemoryHistory()
  const route = '/dogs(modal/Kitty)(panel/Bean)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /birds', () => {
  const history = createMemoryHistory()
  const route = '/birds'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /birds/parrots', () => {
  const history = createMemoryHistory()
  const route = '/birds/parrots'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /birds/parrots', () => {
  const history = createMemoryHistory()
  const route = '/birds/parrots'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /birds/parrots(panel-parrot/Dakota)', () => {
  const history = createMemoryHistory()
  const route = '/birds/parrots(panel-parrot/Dakota)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /dogs(panel-parrot/Dakota)', () => {
  const history = createMemoryHistory()
  const route = '/dogs(panel-parrot/Dakota)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /cats(panel-parrot/Dakota)', () => {
  const history = createMemoryHistory()
  const route = '/cats(panel-parrot/Dakota)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Advanced example /birds/parrots(panel-parrot/Ollie)(modal-cat/)(panel-dog/Bean)', () => {
  const history = createMemoryHistory()
  const route = '/birds/parrots(panel-parrot/Ollie)(modal-cat/)(panel-dog/Bean)'
  history.push(route)

  const tree = renderer
    .create(<Router history={history}>{advancedExample}</Router>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
