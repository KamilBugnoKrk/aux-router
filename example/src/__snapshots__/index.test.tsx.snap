// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced example 1`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
    <button
      onClick={[Function]}
    >
      Birds
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style={
        Object {
          "pointerEvents": "",
          "visibility": undefined,
        }
      }
    >
      <div
        className="panel"
      >
        <img
          alt="Bean"
          src="/aux-router/dog1.png"
        />
        <p>
          Name: 
          Bean
        </p>
        <p>
          Age: 
          5
        </p>
        <button
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Advanced example 2`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
    <button
      onClick={[Function]}
    >
      Birds
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style={
        Object {
          "pointerEvents": "",
          "visibility": undefined,
        }
      }
    >
      <div
        className="panel"
      >
        <img
          alt="Bean"
          src="/aux-router/dog1.png"
        />
        <p>
          Name: 
          Bean
        </p>
        <p>
          Age: 
          5
        </p>
        <button
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Advanced example 3`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
    <button
      onClick={[Function]}
    >
      Birds
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
]
`;

exports[`Advanced example 4`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
    <button
      onClick={[Function]}
    >
      Birds
    </button>
  </nav>,
  <section>
    <h1>
      Dogs
    </h1>
    <ul>
      <li>
        <img
          alt="Bean"
          src="/aux-router/dog1.png"
        />
        <div>
          <h2>
            Bean
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Bella"
          src="/aux-router/dog2.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Axel"
          src="/aux-router/dog3.png"
        />
        <div>
          <h2>
            Axel
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Odie"
          src="/aux-router/dog4.png"
        />
        <div>
          <h2>
            Odie
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Sheba"
          src="/aux-router/dog5.png"
        />
        <div>
          <h2>
            Sheba
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
    </ul>
  </section>,
]
`;

exports[`Advanced example 5`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
    <button
      onClick={[Function]}
    >
      Birds
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
]
`;

exports[`Advanced example 6`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
    <button
      onClick={[Function]}
    >
      Birds
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style={
        Object {
          "pointerEvents": "",
          "visibility": undefined,
        }
      }
    >
      <div
        className="panel"
      >
        <img
          alt="Bean"
          src="/aux-router/dog1.png"
        />
        <p>
          Name: 
          Bean
        </p>
        <p>
          Age: 
          5
        </p>
        <button
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Advanced example 7`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
]
`;

exports[`Advanced example 8`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
  </nav>,
  <section>
    <h1>
      Cats
    </h1>
    <ul>
      <li>
        <img
          alt="Bella"
          src="/aux-router/cat1.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Kitty"
          src="/aux-router/cat2.png"
        />
        <div>
          <h2>
            Kitty
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Charlie"
          src="/aux-router/cat3.png"
        />
        <div>
          <h2>
            Charlie
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Leo"
          src="/aux-router/cat4.png"
        />
        <div>
          <h2>
            Leo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Milo"
          src="/aux-router/cat5.png"
        />
        <div>
          <h2>
            Milo
          </h2>
          <button
            onClick={[Function]}
          >
            Open modal
          </button>
        </div>
      </li>
    </ul>
  </section>,
  <div>
    <div
      className="panel"
    >
      <img
        alt="Bean"
        src="/aux-router/dog1.png"
      />
      <p>
        Name: 
        Bean
      </p>
      <p>
        Age: 
        5
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>,
  <div>
    <div
      className="modal"
    >
      <img
        alt="Kitty"
        src="/aux-router/cat2.png"
      />
      <p>
        Name: 
        Kitty
      </p>
      <p>
        Age: 
        2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>,
]
`;

exports[`Advanced example 9`] = `
Array [
  <nav>
    <button
      onClick={[Function]}
    >
      Cats
    </button>
    <button
      onClick={[Function]}
    >
      Dogs
    </button>
  </nav>,
  <section>
    <h1>
      Dogs
    </h1>
    <ul>
      <li>
        <img
          alt="Bean"
          src="/aux-router/dog1.png"
        />
        <div>
          <h2>
            Bean
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Bella"
          src="/aux-router/dog2.png"
        />
        <div>
          <h2>
            Bella
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Axel"
          src="/aux-router/dog3.png"
        />
        <div>
          <h2>
            Axel
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Odie"
          src="/aux-router/dog4.png"
        />
        <div>
          <h2>
            Odie
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Sheba"
          src="/aux-router/dog5.png"
        />
        <div>
          <h2>
            Sheba
          </h2>
          <button
            onClick={[Function]}
          >
            Open panel
          </button>
        </div>
      </li>
    </ul>
  </section>,
  <div>
    <div
      className="panel"
    >
      <img
        alt="Bean"
        src="/aux-router/dog1.png"
      />
      <p>
        Name: 
        Bean
      </p>
      <p>
        Age: 
        5
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>,
  <div>
    <div
      className="modal"
    >
      <img
        alt="Kitty"
        src="/aux-router/cat2.png"
      />
      <p>
        Name: 
        Kitty
      </p>
      <p>
        Age: 
        2
      </p>
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>,
]
`;

exports[`AuxLink renders correctly 1`] = `
<button
  onClick={[Function]}
>
  Component A - one
</button>
`;

exports[`AuxRoute does not render when path is empty 1`] = `null`;

exports[`AuxRoute renders nested component when path contains correct data 1`] = `
Array [
  <p>
    One
  </p>,
  <p>
    Two
  </p>,
]
`;

exports[`AuxRoute renders nested component when path contains several independent routes 1`] = `
<p>
  Two
</p>
`;

exports[`AuxRoute renders when path contains correct data 1`] = `
<p>
  One
</p>
`;
